### PR TITLE
docs(aio): updated i18n guide and example

### PR DESCRIPTION
**Pull request by ocombe**
## PR Type
What kind of change does this PR introduce?
```
[x] Documentation content changes
```

Target: v5.0.0 docs

## What is the current behavior?
I18n guide is outdated

Issue Number: #19510

## What is the new behavior?
Updated i18n guide for Angular v5+ and cli 1.5+.

## Does this PR introduce a breaking change?
```
[x] No
```

## Things that we could add later
- how to create multiple builds
- how to use the new locale API for libraries/generic components
- how to change base href
- how to update meta data and lang attribute (for html tag)

### CODE REVIEW

File: aio/content/examples/i18n/doc-files/main.2.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// #docregion
+import {enableProdMode, TRANSLATIONS, TRANSLATIONS_FORMAT} from '@angular/core';
```
Comment (Foxandxss) [inline aio/content/examples/i18n/doc-files/main.2.ts]: This and all the other `import {}` needs spaces, AKA `{ enableProdMode }` and not `{enableProdMode}`

Comment (Foxandxss) [reply aio/content/examples/i18n/doc-files/main.2.ts]: Of course, in all files

**Review by Foxandxss — DISMISSED** [review]

**Review by gkalpak — APPROVED** [review]
LGTM as soon as Travis is happy (except for the guide content, which has been approved by @jenniferfell).

**Comment by gkalpak** [comment]
The Travis failures will probably go away with a rebase on master.
Shouldn't this land on 5.0.x as well. If that is the case, I think you need to label it as `target: master and patch`.

File: aio/tools/examples/example-boilerplate.js [hunk 2]
```diff
@@ -31,6 +31,26 @@ const BOILERPLATE_PATHS = {
     'tsconfig.json',
     'tslint.json'
   ],
+  i18n: [
```
Comment (Foxandxss) [inline aio/tools/examples/example-boilerplate.js]: You don't need any of this.

**Review by Foxandxss — DISMISSED** [review]

File: aio/content/examples/i18n/doc-files/app.component.html [hunk 3]
```diff
@@ -29,4 +29,3 @@ <h1 i18n="@@introductionHeader">Hello i18n!</h1>
 <!--#docregion i18n-title-->
 <img [src]="logo" title="Angular logo">
 <!--#enddocregion i18n-title-->
-Contact GitHub API Training Shop Blog About
```
[the remark this answers is not in the document]
Comment (ocombe) [reply aio/content/examples/i18n/doc-files/app.component.html]: yeah, no idea why we had this

File: aio/content/examples/i18n/src/app/app.component.html [hunk 4]
```diff
@@ -17,19 +17,19 @@ <h1 i18n="User welcome|An introduction header for this sample@@introductionHeade
...
+<span i18n>Updated {minutes, plural, =0 {just now} =1 {one minute ago} other {{{minutes}} minutes ago}}</span>
 <!--#enddocregion i18n-plural-->
-({{wolves}})
+({{minutes}})
 <br><br>
-<button (click)="male()">&#9794;</button> <button (click)="female()">&#9792;</button>
+<button (click)="male()">&#9794;</button> <button (click)="female()">&#9792;</button> <button (click)="other()">&#9895;</button>
 <!--#docregion i18n-select-->
-<span i18n>The hero is {gender, select, m {male} f {female}}</span>
+<span i18n>The author is {gender, select, m {male} f {female} o {other}}</span>
```
[the remark this answers is not in the document]
Comment (ocombe) [reply aio/content/examples/i18n/src/app/app.component.html]: this was the example used previously, I just added other, but you're right, maybe I can find another example

File: aio/content/examples/i18n/src/locale/messages.xlf [hunk 5]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2">
+  <file source-language="en" datatype="plaintext" original="ng2.template">
+    <body>
+      <trans-unit id="introductionHeader" datatype="html">
+        <source>
+    Hello i18n!
+</source>
+        <context-group purpose="location">
+          <context context-type="sourcefile">app\app.component.ts</context>
```
[the remark this answers is not in the document]
Comment (ocombe) [reply aio/content/examples/i18n/src/locale/messages.xlf]: no difference, this is purely for documentation (it's generated and not used)

**Comment by vicb** [comment]
@ocombe where to merge ?

**Comment by ocombe** [comment]
should be master & patch apparently

**Pull request merged**